### PR TITLE
Better soundtrack

### PR DESCRIPTION
**Pull request by jcorm714**
- Moved sound effects  and soundtrack to SoundManager Prefab
- Added soundManager to player
- Organized Audio clips in one spot
- Made Network event for changing music work

### CODE REVIEW

File: Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -192,8 +198,8 @@ public void LoadGame()
 
         if (PhotonNetwork.IsMasterClient && PressPlayButtonOnce == false)
         {
-            NetworkSceneChangedRaiseEvent();
             MusicChangeRaiseEvent();
```
Comment (pnaatz) [inline Assets/Scripts/GameManager.cs:201]: are both Raise event functions necessary?

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -55,6 +56,7 @@ private void Awake()
         {
             freeLookCam.SetActive(false);
             GetComponent<AudioListener>().enabled = false;
+            soundManager.isRemotePlayer = true;
```
Comment (pnaatz) [inline Assets/Scripts/Player.cs:59]: good job

File: Assets/Scripts/SoundManager.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    private void Awake()
+    {
+        // when a player is loaded through the game manager it will determine
+        // where audio for that player is routed by default, all remote players are muted
+        // this can be expanded upon later on to utilize unity's 3d audio for player player sfx
+        // when two players are close to eachother. However, there will need to be More Audio Mixer groups
+        AudioMixer mix = Resources.Load("PlayerSounds") as AudioMixer;
+        if (isRemotePlayer)
+        {
+            outputGroup = mix.FindMatchingGroups("RemotePlayer")[0];
```
Comment (pnaatz) [inline Assets/Scripts/SoundManager.cs:57]: either add comments or enum so "0" is known as SFX

File: Assets/Scripts/SoundManager.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+        } else
+        {
+            outputGroup = mix.FindMatchingGroups("LocalPlayer")[0];
+        }
+
+        sfx = mix.FindMatchingGroups($"{outputGroup.name}/SFX")[0];
```
Comment (pnaatz) [inline Assets/Scripts/SoundManager.cs:64]: add comment on why you think its working

File: Assets/Scripts/SoundManager.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+    public void PlayWalking()
+    {
+        return;
+    }
+
+    #endregion
+
+
+    private void OnTriggerEnter(Collider other)
```
Comment (pnaatz) [inline Assets/Scripts/SoundManager.cs:131]: if it doesn't work, get rid of it.

File: Assets/Scripts/SoundManager.cs [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+    private void OnTriggerExit(Collider other)
+    {
+        if (other.gameObject.tag == "Player")
+        {
+            resumeOriginalTrack();
+        }
+    }
+
+    public void resumeOriginalTrack()
```
Comment (pnaatz) [inline Assets/Scripts/SoundManager.cs:148]: get rid of it

File: ProjectSettings/TagManager.asset [hunk 7]
```diff
@@ -11,6 +11,7 @@ TagManager:
   - Room
   - LocationForPickUp
   - PointsPickUp
+  - BackgroundMusic
```
Comment (pnaatz) [inline ProjectSettings/TagManager.asset:14]: remove if not needed

**Review by pnaatz — CHANGES_REQUESTED** [review]
finish changing names of sound files please.

File: Assets/Scripts/PlayerPickUp.cs [hunk 8]
```diff
@@ -81,7 +83,6 @@ void Update()
                     DropOBJ();
                     PickUp = null;
                     isHoldingOBJ = false;
-                    PlayerPickUpSound();
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/PlayerPickUp.cs:84]: Think we should use a different sound than the pickup one

File: Assets/Scripts/Player.cs [hunk 9]
```diff
@@ -20,7 +20,7 @@ public class Player : MonoBehaviourPunCallbacks, IPunObservable
 
     public int score = 0;
     public bool isFiring = false;
-
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/Player.cs:23]: no

File: Assets/Scripts/PlayerPickUp.cs [hunk 10]
```diff
@@ -6,7 +6,6 @@
 using Photon.Realtime;
 using ExitGames.Client.Photon;
 
-[RequireComponent(typeof(AudioSource))]
 public class PlayerPickUp : MonoBehaviourPunCallbacks, IPunObservable, IOnEventCallback
 {
     public GameObject PickUp;
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/PlayerPickUp.cs:11]: did it

File: Assets/Scripts/PlayerPickUp.cs [hunk 11]
```diff
@@ -194,9 +191,11 @@ public void OnEvent(EventData photonEvent)
             PickUp = null;
             isHoldingOBJ = false;
         }
-        else if (eventCode == NetworkCodes.DeleteObjectInDropoffCode)
+        //added this if statement here, because it doesn't want to work in the SoundManager script
+        else if (eventCode == NetworkCodes.ChangeToGameMusicEvent)
         {
-            int id = (int)((object[])photonEvent.CustomData)[0];
+            this.GetComponentInChildren<SoundManager>().SceneTheme.Stop();
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/PlayerPickUp.cs:197]: fixed

File: Assets/Scripts/Player_Movement.cs [hunk 12]
```diff
@@ -3,7 +3,6 @@
 using UnityEngine;
 using Photon.Pun;
 
-[RequireComponent(typeof(AudioSource))]
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/Player_Movement.cs:6]: no

File: Assets/Scripts/Player_Movement.cs [hunk 13]
```diff
@@ -75,17 +74,19 @@ void Update()
 
                     rb.MovePosition(transform.position += moveDir.normalized * speed * Time.deltaTime);
                     Anim.SetBool("IsRunning", true);
-                    Sound.Play(); //play running sound
```
[the remark this answers is not in the document]
Comment (jcorm714) [reply Assets/Scripts/Player_Movement.cs:78]: fixed

File: Assets/Scripts/GameManager.cs (hunk 1 above)
Comment (jcorm714) [reply Assets/Scripts/GameManager.cs:196]: no only the one on line 195 worked

**Comment by jcorm714** [comment]
Added new game song, and changed preview time to 30s to match the length of the intro.

File: Assets/Scripts/PlayerPickUp.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (pnaatz) [reply Assets/Scripts/PlayerPickUp.cs:84]: can you make a sfxManager.PlayerPutDownSound() like you did for the walking sound please

File: Assets/Scripts/SoundManager.cs (hunk 3 above)
Comment (pnaatz) [reply Assets/Scripts/SoundManager.cs:59]: I think you misplaced this comment with the comment requested on line 67.

**Review by pnaatz — CHANGES_REQUESTED** [review]

**Review by pnaatz — APPROVED** [review]

**Pull request merged**